### PR TITLE
Feature[C4R-777] CRUD Practice Programs

### PR DESCRIPTION
**Pull request by birloiflorian**
[C4R-777](https://wearetribus.atlassian.net/browse/C4R-764) - CRUD Practice Program

### CODE REVIEW

File: backend/src/modules/practice-program/services/practice-program.service.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+      let skills = [];
+      if (skillsIds?.length > 0) {
+        skills = await this.nomenclatureService.getSkills({
+          where: {
+            id: In(skillsIds),
+          },
+        });
+      }
+
+      // 6. check if undetermined flad and end date have correct values
```
Comment (dragos1195) [inline backend/src/modules/practice-program/services/practice-program.service.ts:186]: flag*

**Review by dragos1195 — CHANGES_REQUESTED** [review]

File: backend/src/shared/entities/city.entity.ts [hunk 2]
```diff
@@ -34,6 +36,9 @@ export class City {
   @Index()
   county: County;
 
+  @OneToMany(() => PracticeProgram, (practicePrgram) => practicePrgram.location)
```
Comment (radulescuandrew) [inline backend/src/shared/entities/city.entity.ts:39]: E necesar asta aici? 

```If you only care about the @ManyToOne relationship, you can define it without having @OneToMany on the related entity. Where you set @ManyToOne - its related entity will have "relation id" and foreign key.```

**Review by radulescuandrew — COMMENTED** [review]
1 thing

File: backend/src/modules/practice-program/services/practice-program.service.ts (hunk 1 above)
Comment (birloiflorian) [reply backend/src/modules/practice-program/services/practice-program.service.ts]: Fixed

File: backend/src/shared/entities/city.entity.ts (hunk 2 above)
Comment (birloiflorian) [reply backend/src/shared/entities/city.entity.ts]: Nu este necesar, a fost scoasa.

**Pull request merged**